api/src/opentrons/hardware_control/backends/ot3controller.py

```
@@ -380,7 +380,7 @@ async def get_attached_instruments(
         def _synthesize_model_name(
             name: FirmwarePipetteName, model: int
         ) -> "PipetteModel":
-            return cast("PipetteModel", name.name + "_v3." + str(model))
+            return cast("PipetteModel", f"{name.name}_v{model//10}.{model%10}")
```

How does this fix the original problem if the pipette is not flashed? It will still give the incorrect model lookup info if I'm reading this correctly.

---

that's not what solves it. the concept here is that we previously may have wanted to sort of suppress errors coming from unflashed pipettes, but now we can actually flash them, so instead let's emit a structured error. So, it's the code that catches `KeyErrors` from pipette config and turns them into `UnknownPipetteModel` exceptions that solves the issue.

---

Oh the answer to this was: we forgot to commit half the work. Here's the other half that actually does what I said above @Laura-Danielle